config/config.go

```diff
@@ -140,7 +144,11 @@ func (l *Loader) getResolver() FileResolver {
 // YamlProvider returns function to create Yaml based configuration provider
 func (l *Loader) YamlProvider() ProviderFunc {
 	return func() (Provider, error) {
-		return NewYAMLProviderFromFiles(false, l.getResolver(), l.getFiles()...), nil
+		static := NewYAMLProviderFromFiles(false, l.getResolver(), l.getStaticFiles()...)
+		interpolated := NewYAMLProviderWithExpand(false, l.getResolver(), os.LookupEnv, l.getFiles()...)
+
+		// Interpolated files will have higher priority than static.
+		return NewProviderGroup("yaml", static, interpolated), nil
```

Comment:
Yes and the reason is that if you want something to be dynamic, it should have a higher priority(same as dynamic providers have an upper hand over static ones)